src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCaseDefaultColonCheck.java

```diff
@@ -0,0 +1,220 @@
+ *         default : // violation, whitespace before ':' is not allowed here
+ *             break;
+ *     }
+ *
+ *     switch(2) {
+ *         case 2: // ok
+ *             break;
+ *         case 3, 4
+ *                  : break; // violation, whitespace before ':' is not allowed here
+ *         case 4,
+ *               5: break; // ok
+ *         default
+ *               : // violation, whitespace before ':' is not allowed here
+ *             break;
+ *     }
+ *
+ *     switch(day) {
+ *         case MONDAY, FRIDAY, SUNDAY: System.out.println("  6"); break;
+ *         case TUESDAY               : System.out.println("  7"); break; // violation
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code ws.preceded}
+ * </li>
+ * </ul>
+ *
+ * @since 8.44
+ */
+@StatelessCheck
+public class NoWhitespaceBeforeCaseDefaultColonCheck
+    extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "ws.preceded";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.COLON};
+    }
+
+    @Override
+    public boolean isCommentNodesRequired() {
+        return true;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (isInSwitch(ast) && isWhiteSpaceBeforeColon(ast)) {
+            log(ast, MSG_KEY, ast.getText());
+        }
+    }
+
+    /**
+     * Checks if the colon is inside a switch block.
+     *
+     * @param colonAst DetailAST to check.
+     * @return true, if colon case is inside a switch block.
+     */
+    private static boolean isInSwitch(DetailAST colonAst) {
+        return TokenUtil.isOfType(colonAst.getParent(), TokenTypes.LITERAL_CASE,
+                TokenTypes.LITERAL_DEFAULT);
+    }
+
+    /**
+     * Checks if there is a whitespace before the colon of a switch case or switch default.
+     *
+     * @param colonAst DetailAST to check.
+     * @return true, if there is whitespace preceding colonAst.
+     */
+    private static boolean isWhiteSpaceBeforeColon(DetailAST colonAst) {
+        final DetailAST parent = colonAst.getParent();
+        final boolean result;
+        if (isOnDifferentLineWithPreviousToken(colonAst)) {
+            result = true;
+        }
+        else if (parent.getType() == TokenTypes.LITERAL_CASE) {
+            result = isWhitespaceBeforeColonOfCase(colonAst);
+        }
+        else {
+            result = isWhitespaceBeforeColonOfDefault(colonAst);
+        }
+        return result;
+    }
+
+    /**
+     * Checks if there is a whitespace before the colon of a switch case.
```

Comment:
2nd sentence should mention comment support.

Reply:
I think this is not required as comment handling is basic requirement it is being done in both `case` and `default`